### PR TITLE
Adding commands for the worker.

### PR DESCRIPTION
**Pull request by nikolatzotchev**

### CODE REVIEW

File: src/ja/common/message/worker_commands/get_workload.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+This command will get the cpu_load and other information about the state of the work machine
+"""
+from typing import Dict
+
+from ja.common.message.base import Serializable
+from ja.common.message.worker import WorkerCommand, WorkerResponse
+from ja.worker.worker_client import Worker
+
+
+class CancelJobCommand(WorkerCommand):
```
Comment (M1keReck) [inline src/ja/common/message/worker_commands/get_workload.py]: You forgot to change the name of the class to QueryCommand, or whatever you want to call it.

File: src/ja/common/message/worker_commands/set_filters.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+This command will change the filters on the docker event loop
+"""
+from typing import Dict
+
+from ja.common.message.base import Serializable
+from ja.common.message.worker import WorkerCommand, WorkerResponse
+from ja.worker.worker_client import Worker
+
+
+class CancelJobCommand(WorkerCommand):
```
Comment (M1keReck) [inline src/ja/common/message/worker_commands/set_filters.py]: Wrong class name again

File: src/ja/common/message/worker_commands/get_workload.py (hunk 1 above)
Comment (nikolatzotchev) [reply src/ja/common/message/worker_commands/get_workload.py]: I'm not even good at copy pasting.

File: src/ja/common/message/worker_commands/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+"""
+This package contains the worker commands classes
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/__init__.py]: commands classes -> command classes

File: src/ja/common/message/worker_commands/cancel_job.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+"""
+This command will cancel running job an the work machine
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/cancel_job.py]: running job -> a running job
an -> on

File: src/ja/common/message/worker_commands/cancel_job.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        @return: a WorkerResponse with the appropriate response
+        """
+
+    def to_dict(self) -> Dict[str, object]:
+        """!
+        @return: returns a dictionary that represents this object
+        """
+
+    @classmethod
+    def from_dict(cls, property_dict: Dict[str, object]) -> Serializable:
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/cancel_job.py]: Change return type to `"CancelJobCommand"`

File: src/ja/common/message/worker_commands/get_workload.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        @return: a WorkerResponse with the appropriate response
+        """
+
+    def to_dict(self) -> Dict[str, object]:
+        """!
+        @return: returns a dictionary that represents this object
+        """
+
+    @classmethod
+    def from_dict(cls, property_dict: Dict[str, object]) -> Serializable:
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/get_workload.py]: Change return type to `"GetWorkloadCommand"`

File: src/ja/common/message/worker_commands/get_workload.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+This command will get the cpu_load and other information about the state of the work machine
+"""
+from typing import Dict
+
+from ja.common.message.base import Serializable
+from ja.common.message.worker import WorkerCommand, WorkerResponse
+from ja.worker.worker_client import Worker
+
+
+class GetWorkLoadCommand(WorkerCommand):
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/get_workload.py]: GetWorkLoadCommand -> GetWorkloadCommand

File: src/ja/common/message/worker_commands/get_workload.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+    def execute(self, worker_client: Worker) -> WorkerResponse:
+        """!
+        Get the information about the work machine using the provided worker_client
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/get_workload.py]: worker_client -> `@worker_client`

File: src/ja/common/message/worker_commands/set_filters.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+This command will change the filters on the docker event loop
+"""
+from typing import Dict
+
+from ja.common.message.base import Serializable
+from ja.common.message.worker import WorkerCommand, WorkerResponse
+from ja.worker.worker_client import Worker
+
+
+class ChangeFiltersCommand(WorkerCommand):
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/set_filters.py]: What is the purpose of this class?

File: src/ja/common/message/worker_commands/set_filters.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        @return: a WorkerResponse with the appropriate response
+        """
+
+    def to_dict(self) -> Dict[str, object]:
+        """!
+        @return: returns a dictionary that represents this object
+        """
+
+    @classmethod
+    def from_dict(cls, property_dict: Dict[str, object]) -> Serializable:
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/set_filters.py]: Change return type to `"ChangeFiltersCommand"`

File: src/ja/common/message/worker_commands/start_job.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+        @return: a WorkerResponse with the appropriate response
+        """
+
+    def to_dict(self) -> Dict[str, object]:
+        """!
+        @return: returns a dictionary that represents this object
+        """
+
+    @classmethod
+    def from_dict(cls, property_dict: Dict[str, object]) -> Serializable:
```
Comment (JohannesGaessler) [inline src/ja/common/message/worker_commands/start_job.py]: Change return type to `"StartJobCommand"`

File: src/ja/common/message/worker_commands/set_filters.py (hunk 9 above)
Comment (nikolatzotchev) [reply src/ja/common/message/worker_commands/set_filters.py]: We should be able to change the [filters](https://docs.docker.com/engine/reference/commandline/events/), from the server. On what occasions the worker is going to notify the server.

Comment (nikolatzotchev) [reply src/ja/common/message/worker_commands/set_filters.py]: We will probably stick with a fix ones, but nevertheless I decided there could be the possibility of changing them.

**Pull request merged**